runner/execer/os/os_execer_test.go

```diff
@@ -81,6 +83,46 @@ func TestParentProcGroupAndChildren(t *testing.T) {
 	}
 }
 
+func TestAbortCatch(t *testing.T) {
+	e := NewBoundedTestExecer(0, &osProcGetter{})
+	var stdout, stderr bytes.Buffer
+	cmd := execer.Command{
+		Argv:   []string{"sh", "./trap_script.sh"},
+		Stderr: &stderr,
+		Stdout: &stdout,
+	}
+
+	proc, err := e.Exec(cmd)
+	if err != nil {
+		t.Fatal(err)
+	}
+	pid := proc.(*osProcess).cmd.Process.Pid
+
+	time.Sleep(2 * time.Second)
+	usage, err := e.memUsage(pid)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if usage == 0 {
+		t.Fatalf("Expected usage to be >0 for process %d", pid)
+	}
+
+	proc.Abort()
+	usage, err = e.memUsage(pid)
+	if usage != 0 {
+		t.Fatalf("Expected memUsage to be 0 after Abort & Kill, was %d", usage)
+	}
+
+	time.Sleep(3 * time.Second)
+	usage, err = e.memUsage(pid)
+	if err == nil {
+		t.Fatalf("Expected %d to not exist as a process anymore.", pid)
```

Comment:
Note: pids do get reused so we could potentially see a false positive on this condition... But it'll be super rare given the short duration of the sleep. Think it's more important to check for this and possibly run into a false positive than to assume it works as expected.

Reply:
yeah that's fine